### DISCUSSION TITLE
What is the right workflow configuration for my situation?

### DISCUSSION THREAD

**Discussion by Andre601** [crowdin/github-action#128] [Q&A]
I would like to set up two actions to push to or pull from Crowdin under the following situations:

- Push the source file to Crowdin whenever I push to the repository AND the source file has been altered in some way (This is easy enough for me to achieve).
- Pull from Crowdin every 10 minutes and only commit files that have complete translations (No partially translated stuff).

I currently have an open PR to add said actions to my repository [[Link](https://github.com/purrbot-site/PurrBot/pull/488)] and would like feedback on what I need to add if anything to add at all.
Should I add any additional settings to the action to achieve the above goals and conditions?
Like will I need to add `skip_untranslated_files: true` to the actions' `with` options or is that not needed?

I currently have it set as described in the OAuth to Action Migration Guide, so I'm not sure if I also need to alter some other settings in the action (i.e. add the `crowdin_file` option?) or if I have to change some other stuff in the `crowdin.yml`

My Secrets are set to `CROWDIN_ID` with the project ID and `CROWDIN_TOKEN` with the API token I got from `Tools -> Project Id` and `Tools -> API Key` in my Crowdin project respectively.

Some extra goals I would like to achieve, but that aren't as important for me right now:

- Define a PR label, Body and add labels to it.
- Set a Commit message (Preferably with support for placeholders such as `%locale%` but from looking at other repos using this action, this doesn't seem to be supported (yet). See also #45 for more context).

**Comment by Rommka4** [comment]
Hi @Andre601,

Yes, to export fully translated files, you need to add the  `"skip_untranslated_files": true` parameter to your crowdin.yml file. In this case, if the file has at least one untranslated string, it will not be exported. In the article at the link below is more about export options:
https://support.crowdin.com/configuration-file/#export-options

Regarding scheduled synchronization, I need to check with our developers

**Reply by Rommka4** [reply]
Hi @Andre601,

Coming back to you with updates, using GitHub action you can also prescribe a synchronization schedule in the configuration. You can read more at the link below:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

Hope this helps you🙂
